libraries/standard/json/src/json.c

```diff
@@ -364,9 +378,13 @@ static bool_ skipEscape( const char * buf,
                          size_t max )
 {
     bool_ ret = false;
-    size_t i = *start;
+    size_t i;
+
+    assert( ( buf != NULL ) && ( start != NULL ) && ( max > 0 ) );
+
+    i = *start;
 
-    if( ( ( i + 1U ) < max ) && ( buf[ i ] == '\\' ) )
+    if( ( i < ( max - 1 ) ) && ( buf[ i ] == '\\' ) )
```

Comment:
can `max` ever by `0`?

Reply:
There is an assert at the beginning of this function, and every one like it, that requires max > 0.